### PR TITLE
Convert project to pure ESM and update dependencies

### PR DESCRIPTION
**Pull request by logason**
- Convert project to pure ESM: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c
- Upgrade dependencies to get rid of vulnerabilities

### CODE REVIEW

File: Avo.js [hunk 1]
```diff
@@ -170,13 +169,13 @@
...
-    logEventSent: function logEventSent(eventName, eventProperties, userProperties) {
-      const message = "Event Sent:" + eventName + "Event Props:" + JSON.stringify(eventProperties) + "User Props:" + JSON.stringify(userProperties);
+    logEventSent: function logEventSent(eventName, eventProperties, userProperties, groupProperties) {
+      const message = "Event Sent:" + eventName + "Event Props:" + JSON.stringify(eventProperties) + "User Props:" + JSON.stringify(userProperties) + "Group Props:" + JSON.stringify(groupProperties);
 
       if (__AVO_LOGGER__ && __AVO_LOGGER__.logDebug && __AVO_LOGGER__.logDebug(__AVO_ENV__, message)) {
         return
       }
-      typeof console !== 'undefined' && console.log("[avo] Event Sent:", eventName, "Event Props:", eventProperties, "User Props:", userProperties);
+      typeof console !== 'undefined' && console.log("[avo] Event Sent:", eventName, "Event Props:", eventProperties, "User Props:", userProperties, "Group Props:", groupProperties);
```
Comment (bjornj12) [inline Avo.js:178]: why are group properties coming in here now, something that was forgotten?

File: Avo.js [hunk 2]
```diff
@@ -170,13 +169,13 @@
   };
 
   InternalAvoLogger = {
-    logEventSent: function logEventSent(eventName, eventProperties, userProperties) {
-      const message = "Event Sent:" + eventName + "Event Props:" + JSON.stringify(eventProperties) + "User Props:" + JSON.stringify(userProperties);
+    logEventSent: function logEventSent(eventName, eventProperties, userProperties, groupProperties) {
+      const message = "Event Sent:" + eventName + "Event Props:" + JSON.stringify(eventProperties) + "User Props:" + JSON.stringify(userProperties) + "Group Props:" + JSON.stringify(groupProperties);
```
Comment (bjornj12) [inline Avo.js:173]: why are group properties coming in here now, something that was forgotten?

**Review by bjornj12 — APPROVED** [review]
Agree with the major update. 
Tested the client on our repo and it worked fine! LGTM

File: Avo.js [hunk 3]
```diff
@@ -270,484 +269,519 @@ _avo_invoke_meta = function _avo_invoke_meta(env, type, messages, origin) {
...
+      "email",
+      "authenticationMethod"
+    ]));
+    // debug console in Avo
     if (!__AVO_NOOP__) {
-      if (__AVO_ENV__ === 'prod') {
-      }
-      if (__AVO_ENV__ === 'dev') {
-      }
+      _avo_invoke(__AVO_ENV__, "54e92613-090c-4f0b-afeb-ed720eff3422", "2fec10106ad560f43f307ffdc820e3bd7a474585f9c62ee39961730c35694cb1", messages.map(function(m) { return Object.assign({}, {tag: m.tag, propertyId: m.propertyId, additionalProperties: m.additionalProperties, actualType: m.actualType}); }), 'event');
```
Comment (hugihlynsson) [inline Avo.js:594]: What magic strings are these and why are they changing?

File: Avo.js [hunk 4]
```diff
@@ -757,407 +791,380 @@ _avo_invoke_meta = function _avo_invoke_meta(env, type, messages, origin) {
...
+  signInFailed: signInFailed,
+  cliInvoked: cliInvoked,
+  cliInstalled: cliInstalled,
+  cliConflictResolveAttempted: cliConflictResolveAttempted,
+  cliConflictResolveFailed: cliConflictResolveFailed,
+  cliConflictResolveSucceeded: cliConflictResolveSucceeded,
+  setSystemProperties: setSystemProperties,
+  initAvo: initAvo,
+  avoInspectorApiKey: avoInspectorApiKey,
+}
```
Comment (hugihlynsson) [inline Avo.js:1167]: Do we need to `default` export this map when we've already exported each function individually?

File: Avo.js (hunk 1 above)
Comment (logason) [reply Avo.js:178]: We haven't pulled Avo in this project since we launched Group properties 🚀

File: Avo.js (hunk 3 above)
Comment (logason) [reply Avo.js:594]: The first one is the `eventId` in Avo, the second one is the `eventHash` which we use in the implementation status to confirm if we've seen this exact version of the event sent or not.

File: Avo.js (hunk 4 above)
Comment (logason) [reply Avo.js:1167]: Good question, I'm not sure! Something we can look into in the JS codegen for Avo Functions (this is an Avo generated file)

File: Avo.js (hunk 2 above)
Comment (logason) [reply Avo.js:173]: We haven't pulled Avo in this project since we launched Group properties 🚀

File: Avo.js (hunk 3 above)
Comment (hugihlynsson) [reply Avo.js:594]: Ahh, I see!

File: Avo.js (hunk 4 above)
Comment (hugihlynsson) [reply Avo.js:1167]: Ohh of course!

**Pull request merged**